src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/repository/StaffRepository.kt

```diff
@@ -9,7 +10,19 @@ import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.Staff
 @Repository
 interface StaffRepository : JpaRepository<Staff, Long> {
   fun findByStaffIdentifier(staffIdentifier: Long): CommunityOffenderManager?
-  fun findByStaffIdentifierOrUsernameIgnoreCase(staffIdentifier: Long, username: String): List<CommunityOffenderManager>?
+
+  @Query(
+    """
+        SELECT s FROM CommunityOffenderManager s
+            WHERE s.staffIdentifier = :staffIdentifier OR UPPER(s.username) = UPPER(:username)
+            ORDER BY s.lastUpdatedTimestamp DESC
+    """,
+  )
+  fun findCommunityOffenderManager(
+    staffIdentifier: Long,
+    username: String,
+  ): List<CommunityOffenderManager>
+
```

Comment:
Could have just change to JPQL from a derived query to get the order correct. I original changed this because I was the discriminator would not work on the return type, this I'm sure used to be the case but not with the latest JPA/Spring we are ok!
Could change it to
`findByStaffIdentifierOrUsernameIgnoreCaseOrderByLastUpdatedTimestampDesc` 
if you wish,.. seems a mouth full